firebase-vscode/webpack.common.js

```diff
@@ -0,0 +1,199 @@
+
+/**@type {import('webpack').Configuration}*/
+const extensionConfig = {
+  name: "extension",
+  target: "node", // vscode extensions run in webworker context for VS Code web 📖 -> https://webpack.js.org/configuration/target/#target
+
+  entry: "./src/extension.ts", // the entry point of this extension, 📖 -> https://webpack.js.org/configuration/entry-context/
+  output: {
+    // the bundle is stored in the 'dist' folder (check package.json), 📖 -> https://webpack.js.org/configuration/output/
+    path: path.resolve(__dirname, "dist"),
+    filename: "extension.js",
+    libraryTarget: "commonjs2",
+    devtoolModuleFilenameTemplate: "../[resource-path]",
+  },
+  devtool: "source-map",
+  externals: {
+    vscode: 'commonjs vscode' // the vscode-module is created on-the-fly and must be excluded. Add other modules that cannot be webpack'ed, 📖 -> https://webpack.js.org/configuration/externals/
+  },
+  resolve: {
+    // support reading TypeScript and JavaScript files, 📖 -> https://github.com/TypeStrong/ts-loader
+    // mainFields: ['browser', 'module', 'main'], // look for `browser` entry point in imported node modules
+    mainFields: ["main", "module"],
+    extensions: [".ts", ".js"],
+    alias: {
+      // provides alternate implementation for node module and source files
+      "proxy-agent": path.resolve(__dirname, 'src/stubs/empty-class.js'),
+      "marked-terminal": path.resolve(__dirname, 'src/stubs/empty-class.js'),
+      // "ora": path.resolve(__dirname, 'src/stubs/empty-function.js'),
+      "commander": path.resolve(__dirname, 'src/stubs/empty-class.js'),
+      "inquirer": path.resolve(__dirname, 'src/stubs/inquirer-stub.js'),
+      // This is used for Github deploy to hosting - will need to restore
+      // or find another solution if we add that feature.
+      "libsodium-wrappers": path.resolve(__dirname, 'src/stubs/empty-class.js'),
+      "marked": path.resolve(__dirname, 'src/stubs/marked.js')
+    },
+    fallback: {
+      // Webpack 5 no longer polyfills Node.js core modules automatically.
+      // see https://webpack.js.org/configuration/resolve/#resolvefallback
+      // for the list of Node.js core module polyfills.
+    },
+  },
+  module: {
+    rules: [
+      {
+        test: /\.ts$/,
+        exclude: [/node_modules/],
+        use: [
+          {
+            loader: "ts-loader",
+          },
+        ],
+      },
+      {
+        test: /\.ts$/,
+        loader: "string-replace-loader",
+        options: {
+          multiple: [
+            {
+              search: /(\.|\.\.)[\.\/]+templates/g,
+              replace: "./templates",
+            },
+            {
+              search: /(\.|\.\.)[\.\/]+schema/g,
+              replace: "./schema",
+            },
+            {
+              search: /Configstore\(pkg\.name\)/g,
+              replace: "Configstore('firebase-tools')",
+            },
+            // TODO(hsubox76): replace with something more robust
```

Comment:
I'm trying to avoid changing any CLI code at the moment, this is a stopgap for now to prevent complete breakage. I'll eventually have to go into the CLI code (src/deploy/lifecycleHooks.ts) and figure out a way to bundle `cross-env-shell.js` and run it so that the predeploy can work in windows, and then somehow call the bundled version from the extension. In this initial PR I'm trying to avoid making any changes that could potentially affect the functionality of the CLI.